client/rpc-spec-v2/src/chain_head/subscription/inner.rs

```diff
@@ -107,6 +108,75 @@ impl BlockStateMachine {
 	}
 }
 
+/// Limit the number of ongoing operations across methods.
+#[derive(Clone)]
+struct LimitOperations {
+	/// Maximum number of operations that can be executed at the same time.
+	max_operations: usize,
+	/// The number of ongoing operations for this subscription.
+	ongoing: Arc<Mutex<usize>>,
```

Comment:
maybe you could replace this by [tokio::sync::Semaphore](https://docs.rs/tokio/latest/tokio/sync/struct.Semaphore.html) to avoid implement similar functionality here

Reply:
That makes sense, have wrapped the `tokio::sync::Semaphore` to rely on its permits! Thanks for the suggestion!